src/main/java/uk/gov/hmcts/cmc/claimstore/services/OffersService.java

```diff
@@ -29,9 +32,35 @@ public OffersService(
     }
 
     public void makeOffer(Claim claim, Offer offer, MadeBy party) {
+        assertSettlementIsNotReached(claim);
+
         Settlement settlement = claim.getSettlement().orElse(new Settlement());
         settlement.makeOffer(offer, party);
         offersRepository.updateSettlement(claim.getId(), jsonMapper.toJson(settlement));
         eventProducer.createOfferMadeEvent(claim);
     }
+
+    public void accept(Claim claim, MadeBy party) {
+        assertSettlementIsNotReached(claim);
+
+        Settlement settlement = claim.getSettlement().orElse(new Settlement());
```

Comment:
Accept is not possible on a new (empty) settlement, `orElse` should be removed and the assertion should cover this.

Reply:
changed